x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -26,25 +25,29 @@ func (k Keeper) GetDataCommitmentConfirm(
 	return &confirm
 }
 
-// // GetDataCommitmentConfirmsByCommitment Returns data commitment confirms by commitment
-// func (k Keeper) GetDataCommitmentConfirmsByCommitment(
-// 	ctx sdk.Context,
-// 	commitment string,
-// ) (confirms []types.MsgDataCommitmentConfirm) {
-// 	prefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.DataCommitmentConfirmKey))
-// 	start, end := prefixRange([]byte(commitment)) // How does this work?
-// 	iterator := prefixStore.Iterator(start, end)
+// GetDataCommitmentConfirmsByCommitment Returns data commitment confirms by commitment
+func (k Keeper) GetDataCommitmentConfirmsByCommitment(
+	ctx sdk.Context,
+	commitment string,
+) (confirms []types.MsgDataCommitmentConfirm) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := store.Iterator(nil, nil) // Can we make this faster?
```

Comment:
For now, this is iterating over the whole store and skipping elements it cannot unmarshall.
Any idea on how to make this faster ? Or maybe have some sort of indexing...